api/views/group.py

```diff
@@ -67,3 +73,34 @@ def confirm_ready_to_start(self, request: Request, pk: int) -> Response:  # noqa
         group = self.get_object()
         GroupProcessor.confirm_ready_to_start(group)
         return Response(status=status.HTTP_200_OK)
+
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(
+                name="discard_reason",
+                type=str,
+                enum=[e.value for e in GroupDiscardReason],
+                required=True,
+            )
+        ],
+        responses={
+            status.HTTP_204_NO_CONTENT: OpenApiResponse(description="Group is discarded"),
+            status.HTTP_409_CONFLICT: OpenApiResponse(
+                response=BaseAPIExceptionSerializer,
+                description="Group is not in the pending state",
+            ),
+            status.HTTP_400_BAD_REQUEST: OpenApiResponse(
+                response=ValidationErrorSerializer,
+                description="Something is wrong with the data",
+            ),
+        },
+    )
+    @action(detail=True, methods=["delete"])
+    def discard(self, request: Request, pk: int) -> Response:  # noqa: ARG002
+        group = self.get_object()
+        if group.project_status != GroupProjectStatus.PENDING:
```

Comment:
I also assumed that we can discard it only if a group in the pending state

Reply:
That is correct understanding :)